doc/src/tutorial/assumptions.rst

```diff
@@ -0,0 +1,1836 @@
+.. _assumptions:
+
+=============
+ Assumptions
+=============
+
+The previous page of the tutorial on simplification briefly mentioned setting
+assumptions on symbols in SymPy. This page explains how the assumptions
+systems is used and what the different assumptions predicates mean.
+
+.. note:: This page describes the "old assumptions" system. There is also a
+          "new assumptions" system which is described at the end of this page.
+
+Firstly we consider what happens when taking the square root of the square of
+a concrete integer such as $2$ or $-2$:
+
+    >>> from sympy import sqrt
+    >>> sqrt(2**2)
+    2
+    >>> sqrt((-2)**2)
+    2
+    >>> x = 2
+    >>> sqrt(x**2)
+    2
+    >>> sqrt(x**2) == x
+    True
+    >>> y = -2
+    >>> sqrt(y**2) == y
+    False
+    >>> sqrt(y**2) == -y
+    True
+
+What these examples demonstrate is that for a positive number $x$ we have
+$\sqrt{x^2} = x$ whereas for a negative number we would instead have
+$\sqrt{x^2} = -x$. That may seem obvious but the situation can be more
+surprising when working with a symbol rather then an explicit number. For
+example
+
+    >>> from sympy import Symbol, simplify
+    >>> x = Symbol('x')
+    >>> sqrt(x**2)
+    sqrt(x**2)
+
+It might look as if that should simplify to ``x`` but it does not even if
+:func:`~.simplify` is used:
+
+    >>> simplify(sqrt(x**2))
+    sqrt(x**2)
+
+This is because SymPy will refuse to simplify this expression if the
+simplification is not valid for *every* possible value of ``x``. By default
+the symbol ``x`` is considered only to represent something roughly like an
+arbitrary complex number and the obvious simplification here is only valid for
+positive real numbers. Since ``x`` is not known to be positive or even real no
+simplification of this expression is possible.
+
+We can tell SymPy that a symbol represents a positive real number when
+creating the symbol and then the simplification will happen automatically:
+
+    >>> y = Symbol('y', positive=True)
+    >>> sqrt(y**2)
+    y
+
+This is what is meant by "assumptions" in SymPy. If the symbol ``y``
+is created with ``positive=True`` then SymPy will *assume* that it represents
+a positive real number rather than an arbitrary complex or possibly infinite
+number. That *assumption* can make it possible to simplify expressions or
+might allow other manipulations to work. It is usually a good idea to be as
+precise as possible about the assumptions on a symbol when creating it.
+
+
+The (old) assumptions system
```

Comment:
This is part of a bigger discussion, but I don't think we should say "old" here. Our usage of "old" and "new" is sparse in the docs, enough that people are generally confused about which is which. But I think instead of trying to document it better, we should try to stop using those names, because

1. The "new" assumptions aren't even that new any more,
2. If we do end up merging them, then what would it be? "old-new"?
3. They are only descriptive for what they were historically.

Rather, we should have some better names. I'm not sure what the best ones are just yet. Something like "predicate assumptions" or "expression" assumptions for the new assumptions.

Reply:
I agree that we should use better names for this but if we invent new terms then we also need to have clear references linking those to the old terms.

I would go with "core assumptions" for the old assumptions.